test/tests/pbs_smoketest.py

```diff
@@ -1462,3 +1462,30 @@ def test_jobscript_max_size(self):
                           e.msg[0])
         self.server.log_match("Req;req_reject;Reject reply code=15175",
                               max_attempts=5)
+
+    @checkModule("pbs")
+    def test_import_pbs_module(self):
+        """
+        Test that the pbs module located in the PBS installation directory is
+        able to be loaded and symbols within it accessed.  This test requires
+        the PYTHONPATH environment variable contain the path
+        PBS_EXEC/{lib,lib64}/python/altair.
+        """
+        import pbs
+        msg = "pbs.JOB_STATE_RUNNING=%s" % (pbs.JOB_STATE_RUNNING,)
+        self.logger.info(msg)
+
+    @checkModule("pbs_ifl")
+    def test_import_pbs_ifl_module(self):
+        """
+        Test that the pbs_ifl module located in the PBS installation directory
+        is able to be loaded and a connection to the server can be established.
+        This test requires the PYTHONPATH environment variable contain the path
+        PBS_EXEC/{lib,lib64}/python/altair.
+        """
+        import pbs_ifl
+        server_conn = pbs_ifl.pbs_connect(None)
+        server_stat = pbs_ifl.pbs_statserver(server_conn, None, None)
+        pbs_ifl.pbs_disconnect(server_conn)
+        msg = "server name is %s" % (server_stat.name,)
+        self.logger.info(msg)
```

Comment:
@checkmodule() test may not be reliable,  so I suggest doing:
```
 diff -c pbs_smoketest.py.orig pbs_smoketest.py
*** pbs_smoketest.py.orig       2020-09-15 01:29:21.387629113 -0400
--- pbs_smoketest.py    2020-09-15 01:29:34.083629310 -0400
***************
*** 1463,1469 ****
          self.server.log_match("Req;req_reject;Reject reply code=15175",
                                max_attempts=5)
  
-     @checkModule("pbs")
      def test_import_pbs_module(self):
          """
          Test that the pbs module located in the PBS installation directory is
--- 1463,1468 ----
***************
*** 1471,1481 ****
          the PYTHONPATH environment variable contain the path
          PBS_EXEC/{lib,lib64}/python/altair.
          """
          import pbs
          msg = "pbs.JOB_STATE_RUNNING=%s" % (pbs.JOB_STATE_RUNNING,)
          self.logger.info(msg)
  
-     @checkModule("pbs_ifl")
      def test_import_pbs_ifl_module(self):
          """
          Test that the pbs_ifl module located in the PBS installation directory
--- 1470,1482 ----
          the PYTHONPATH environment variable contain the path
          PBS_EXEC/{lib,lib64}/python/altair.
          """
+         sys.path.append(os.path.join(self.server.pbs_conf['PBS_EXEC'], 'lib',
+                              'python', 'altair'))
          import pbs
          msg = "pbs.JOB_STATE_RUNNING=%s" % (pbs.JOB_STATE_RUNNING,)
          self.logger.info(msg)
  
      def test_import_pbs_ifl_module(self):
          """
          Test that the pbs_ifl module located in the PBS installation directory
***************
*** 1483,1488 ****
--- 1484,1492 ----
          This test requires the PYTHONPATH environment variable contain the path
          PBS_EXEC/{lib,lib64}/python/altair.
          """
+         sys.path.append(os.path.join(self.server.pbs_conf['PBS_EXEC'], 'lib',
+                              'python', 'altair'))
          import pbs_ifl
          server_conn = pbs_ifl.pbs_connect(None)
          server_stat = pbs_ifl.pbs_statserver(server_conn, None, None)
```

Reply:
I'm more than happy to make the changes you suggest; however, I must ask, if @checkModule() is unreliable. then shouldn't it be either fixed or eliminated.  I ask because I used other test code that made use of this decorator as an example, so any issues with the decorator may extend beyond the tests I added.

Reply:
@toonen : I would have to check what's going on with @checkModule. Maybe it's assuming that the environment where pbs_benchpress is invoked already contains path $PBS_EXEC/lib/python/altair.  Maybe I didn't have that in my path when I invoked the testsuite, and it kept failing on @checkModule. When took out that check, and explictly added the path in the test case itself, it worked. This sounds more like a fail-safe way of running the test...

Reply:
@bayucan: I applied the suggested patch in the following commit: 3a61204951b88280b82cc47f5d9ab9912ab31c6c.  Thanks for the suggestion.

Reply:
This had the side effect of identifying an issue with -fsanitize=address, which is currently resulting in the import tests failing when that flag is included.  I'm working on an autotools appropriate way to address the issue, which only reminds me that I'm a decade or so behind on frequent autotools use. :-)